### PR TITLE
Add GT tone map operator

### PR DESCRIPTION
**Pull request by meshula**
Here's a PR with only the new tone map operator. You can learn more about it here - https://www.shadertoy.com/view/Xstyzn

### CODE REVIEW

File: src/operators/gran_turismo.h [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+    Tone Mapper is provided under the MIT License.
+    See the LICENSE.txt file for the conditions of the license. 
+*/
+
+#pragma once
+
+#include <tonemap.h>
+
+#define glsl(x) #x
```
Comment (tizian) [inline src/operators/gran_turismo.h:16]: This trick does not seem to be portable, or at least it doesn't compile on my version of clang on macOS.
I changed all other GLSL code in operators to the C++ multiline string literals though, that should be a less painful way (and really I should have done this from the start)

e.g. 
```C++
R"glsl(
#version 330
int main() {
    out_color = vec4(1);
}
)glsl"
```

Comment (meshula) [reply src/operators/gran_turismo.h:16]: That's cool, I'll replace the use of the macro everywhere I use it; not just here :)

**Comment by tizian** [comment]
Hi Nick,
I went ahead and added this operator to the master branch with the change suggested above. I also found a small typo in the computation of `l0` that caused the `l` parameter to be mostly useless. (The same issue is present on your shadertoy implementation btw.)

Further I realized that both `L0` and `L1` are actually unused –– do you have any insight behind what they represent?

Best,
Tizian

**Comment by meshula** [comment]
Hi, thanks for the review and catching the typo. Have a look here, for L0 and L1. https://www.desmos.com/calculator/gslcdxvipg They don't have anything to do with the final calculation, I left them in because they are in the source material, and I thought others might want to play with a complete representation of the original paper. Maybe @romainguy has some insight, as I worked from his notebook to create the shader.

**Comment by romainguy** [comment]
My Desmos notebook is a straight copy of the original by Polyphony Digital. I only translated the Japanese bits to English :)

**Comment by meshula** [comment]
FWIW, I double checked the translation, and I get the same results as Romain. I think there might be errors or omissions in the original slides. I haven't contacted the original author.

**Comment by tizian** [comment]
Ok, thanks for clarifying @romainguy –– and thanks again for the nice PR @meshula !

**Pull request closed without merging**